src/sas/qtgui/Utilities/OrientationViewer/OrientationViewerController.py

```diff
@@ -0,0 +1,107 @@
+from typing import NamedTuple
+
+from PyQt5 import QtWidgets
+from PyQt5.QtCore import pyqtSignal
+
+
+
+from sas.qtgui.Utilities.UI.OrientationViewerControllerUI import Ui_OrientationViewierControllerUI
+
+class Orientation(NamedTuple):
+    """ Data sent when updating the plot"""
+    theta: int = 0
+    phi: int = 0
+    psi: int = 0
+    dtheta: int = 0
+    dphi: int = 0
+    dpsi: int = 0
+
+
+
+class OrientationViewierController(QtWidgets.QDialog, Ui_OrientationViewierControllerUI):
+
+    """ Widget that controls the orientation viewer"""
+
+    sliderSet = pyqtSignal(Orientation, name='sliderSet')
+    sliderMoved = pyqtSignal(Orientation, name='sliderMoved')
+
+    def __init__(self, parent=None):
+        super().__init__()
+
+        self.setupUi(self)
+
+        self.setLabels(Orientation())
+
+        # All sliders emit the same signal - the angular coordinates in degrees
+        self.thetaSlider.sliderReleased.connect(self.onSliderSet)
+        self.phiSlider.sliderReleased.connect(self.onSliderSet)
+        self.psiSlider.sliderReleased.connect(self.onSliderSet)
+        self.deltaTheta.sliderReleased.connect(self.onSliderSet)
+        self.deltaPhi.sliderReleased.connect(self.onSliderSet)
+        self.deltaPsi.sliderReleased.connect(self.onSliderSet)
+
+        self.thetaSlider.valueChanged.connect(self.onSliderMoved)
+        self.phiSlider.valueChanged.connect(self.onSliderMoved)
+        self.psiSlider.valueChanged.connect(self.onSliderMoved)
+        self.deltaTheta.valueChanged.connect(self.onSliderMoved)
+        self.deltaPhi.valueChanged.connect(self.onSliderMoved)
+        self.deltaPsi.valueChanged.connect(self.onSliderMoved)
+
+    def setLabels(self, orientation: Orientation):
+
+        self.thetaNumber.setText(f"{orientation.theta}°")
+        self.phiNumber.setText(f"{orientation.phi}°")
+        self.psiNumber.setText(f"{orientation.psi}°")
+
+        self.deltaThetaNumber.setText(f"{orientation.dtheta}°")
+        self.deltaPhiNumber.setText(f"{orientation.dphi}°")
+        self.deltaPsiNumber.setText(f"{orientation.dpsi}°")
+
+
+    def onSliderSet(self):
+        theta = self.thetaSlider.value()
+        phi = self.phiSlider.value()
+        psi = self.psiSlider.value()
+
+        dtheta = self.deltaTheta.value()
+        dphi = self.deltaPhi.value()
+        dpsi = self.deltaPsi.value()
+
+        orientation = Orientation(theta, phi, psi, dtheta, dphi, dpsi)
```

Comment:
Repeated below - create a separate method or property, maybe? Also, you are creating a new Orientation object with every move. Is there any noticeable performance difference between updating the existing object vs creating a new one?

Reply:
Yeah, there only needs to be one of them `onSliderSetOrMoved`

There's nothing noticeably slow about the response. As a rule of thumb, the use of "mutable" objects as signals should be avoided. In fact, mutating objects should really be avoided everywhere it is possible to do so, even though it is generally slower. A bit like putting duplicated code in its own function, its slower, but better software engineering practice.